src/awkward/contents/recordarray.py

```diff
@@ -411,6 +411,17 @@ def content(self, index_or_field: str | SupportsIndex) -> Content:
         else:
             return out[: self._length]
 
+    def maybe_content(self, index_or_field) -> Content:
+        if self.has_field(index_or_field):
+            return self.content(index_or_field)
+        else:
+            return ak.contents.IndexedOptionArray(
+                ak.index.Index64(
+                    self._backend.index_nplike.full(self.length, -1, dtype=np.int64)
+                ),
+                ak.contents.EmptyArray(),
+            )
+
```

Comment:
I think this is simple enough and useful enough to warrant inclusion. Easily moved on this, though.

Reply:
Yeah, that's a nice way to generalize.